homeassistant/components/picnic/sensor.py

```diff
@@ -62,14 +64,23 @@ def __init__(
         self._attr_unique_id = f"{config_entry.unique_id}.{description.key}"
 
     @property
-    def native_value(self) -> StateType:
+    def native_value(self) -> StateType | datetime:
         """Return the state of the entity."""
         data_set = (
             self.coordinator.data.get(self.entity_description.data_type, {})
             if self.coordinator.data is not None
             else {}
         )
-        return self.entity_description.value_fn(data_set)
+        value = self.entity_description.value_fn(data_set)
+
+        if not value or self.device_class != SensorDeviceClass.TIMESTAMP:
+            return value
+
+        # Convert to datetime object if the device class is timestamp
+        try:
+            return datetime.strptime(str(value), DATE_TIME_FORMAT)
+        except ValueError:
+            return None
```

Comment:
Add:  `from homeassistant.util import dt as dt_util`

```suggestion
        return dt_util.parse_datetime(value)
```

Reply:
Would be nice to move this to the entity description `value_fn`.

Reply:
I've now used the `homeassistant.util.dt.parse_datetime`, but moving it to `value_fn` was a bit hard as it's needed to differentiate between a None value as returned from the api client or a None value due to a malformed date string.

Reply:
A None value returned from a successful api call should not make the entity unavailable. Unavailable means we failed to contact the api to get a value. If a value is None or missing after a successful api call we should set unknown state.

Reply:
Ok, changed it now that None values after a successful fetch are unknown (for all sensors) and moved the datetime parse to `value_fn`.